File: examples/metal/gateway/example_2/python/__main__.py
import pulumi
import pulumi_equinix as equinix

test = equinix.metal.Vlan("test",
    description="test VLAN in SV",
    metro="sv",
    project_id=project_id)
test1 = equinix.metal.ReservedIpBlock("test1",
    project_id=project_id,
    metro="sv",
    quantity=8)
test_gateway = equinix.metal.Gateway("testGateway",
    project_id=project_id,
    vlan_id=test.id,
    ip_reservation_id=test_equinix_metal_reserved_ip_block["id"])

File: examples/metal/gateway/example_2/go/main.go
package main

import (
	"github.com/equinix/pulumi-equinix/sdk/go/equinix/metal"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		test, err := metal.NewVlan(ctx, "test", &metal.VlanArgs{
			Description: pulumi.String("test VLAN in SV"),
			Metro:       pulumi.String("sv"),
			ProjectId:   pulumi.Any(projectId),
		})
		if err != nil {
			return err
		}
		_, err = metal.NewReservedIpBlock(ctx, "test1", &metal.ReservedIpBlockArgs{
			ProjectId: pulumi.Any(projectId),
			Metro:     pulumi.String("sv"),
			Quantity:  pulumi.Int(8),
		})
		if err != nil {
			return err
		}
		_, err = metal.NewGateway(ctx, "testGateway", &metal.GatewayArgs{
			ProjectId:       pulumi.Any(projectId),
			VlanId:          test.ID(),
			IpReservationId: pulumi.Any(testEquinixMetalReservedIpBlock.Id),
		})
		if err != nil {
			return err
		}
		return nil
	})
}
